### DISCUSSION TITLE
ArgoCD can't connect GCP artifact registry

### DISCUSSION THREAD

**Discussion by ironman-gf** [argoproj/argo-cd#12924] [General]
ArgoCD version: V2.5.2
We already have a helm chart which located on GCP artifact registry, also this helm chart works well for below command:
 helm install ane-gke oci://asia-docker.pkg.dev/xxxxx/xxxx--version 0.2.6
But I ran into some issue when I try to setup a new application base on the helm repo that is GCP artifact registry.

"Unable to create application: application spec for my-app is invalid: InvalidSpecError: repository not accessible: repo client error while testing repository: rpc error: code = Unknown desc = error testing repository connectivity: Get "https://console.cloud.google.com/artifacts/docker/xxxxxasia/axxxxx/index.yaml": Forbidden"

It seems the ArgoCD will fetch the helm index.yaml file to identify the helm chart, but we don't have this index.yaml file in our GCP artifact registry, is this caused the issue? Any solution for this?

Thanks.

**Comment by nissessenap** [comment]
I would argue that using OCI to package your helm chart is the solution/workaround for this. Sadly, the OCI support in ArgoCD is lacking https://github.com/argoproj/argo-cd/issues/10823.
It is possible if you use the old solution with Chart.yaml pointing to an OCI repo, but it's not supported if you try to go directly to OCI chart if I have understood it correctly.

**Comment by lhriley** [comment]
This is my partial workaround so far. This allows us to use OCI charts as dependencies, but I'm still working on how to get argocd to install from our Google Artifact Registry directly:

https://github.com/argoproj/argo-cd/issues/12436#issuecomment-1505879515
